module/data/active-effect/enchantment.mjs

```diff
@@ -0,0 +1,67 @@
+import { DamageData } from "../shared/damage-field.mjs";
+
+/**
+ * System data model for enchantment active effects.
+ */
+export default class EnchantmentData extends foundry.abstract.TypeDataModel {
+  /** @override */
+  static defineSchema() {
+    return {};
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle enchantment-specific changes to the item.
+   * @param {Item5e} item                The Item to whom this effect should be applied.
+   * @param {EffectChangeData} change    The change data being applied.
+   * @param {Record<string, *>} changes  The aggregate update paths and their updated values.
+   * @returns {boolean|void}             Return false to prevent normal application from occurring.
+   */
+  _applyLegacy(item, change, changes) {
+    console.log("EnchantmentData#_applyLegacy", item, change, changes);
+    let key = change.key.replace("system.", "");
+    switch (change.key) {
+      case "system.attack.bonus":
+      case "system.attack.flat":
+        for ( const activity of item.system.activities?.getByTypes("attack") ?? [] ) {
+          changes[`system.activities.${activity.id}.${key}`] = ActiveEffect.applyField(activity, { ...change, key });
+        }
+        return false;
+      case "system.damage.parts":
+        try {
+          let damage;
+          const parsed = JSON.parse(change.value);
+          if ( foundry.utils.getType(parsed) === "Object" ) damage = new DamageData(parsed);
+          else damage = new DamageData({ custom: { enabled: true, formula: parsed[0][0] }, types: [parsed[0][1]] });
+          for ( const activity of item.system.activities?.getByTypes("attack", "damage", "save") ?? [] ) {
+            const value = damage.clone();
+            value.enchantment = true;
+            value.locked = true;
+            changes[`system.activities.${activity.id}.damage.parts`] = ActiveEffect.applyField(
+              activity, { ...change, key: "damage.parts", value: value }
+            );
+          }
+          return false;
+        } catch(err) {
+          console.log(err);
```

Comment:
Should we return false here as well if an error is thrown?

Reply:
My thought was just to fall back to the standard behavior and see if it can make sense of it.